packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -613,36 +618,68 @@ class _FocusableActionDetectorState extends State<FocusableActionDetector> {
   void _handleMouseEnter(PointerEnterEvent event) {
     assert(widget.onShowHoverHighlight != null);
     if (!_hovering) {
-      // TODO(gspencergoog): remove scheduleMicrotask once MouseRegion event timing has changed.
-      scheduleMicrotask(() { setState(() { _hovering = true; _handleShowHoverHighlight(); }); });
+      _mayTriggerCallback(task: () {
+        setState(() { _hovering = true; });
+      });
     }
   }
 
   void _handleMouseExit(PointerExitEvent event) {
     assert(widget.onShowHoverHighlight != null);
     if (_hovering) {
-      // TODO(gspencergoog): remove scheduleMicrotask once MouseRegion event timing has changed.
-      scheduleMicrotask(() { setState(() { _hovering = false; _handleShowHoverHighlight(); }); });
+      _mayTriggerCallback(task: () {
+        setState(() { _hovering = false; });
+      });
     }
   }
 
   bool _focused = false;
   void _handleFocusChange(bool focused) {
     if (_focused != focused) {
-      setState(() {
-        _focused = focused;
-        _handleShowFocusHighlight();
-        widget.onFocusChange?.call(_focused);
+      _mayTriggerCallback(task: () {
+        setState(() {
+          _focused = focused;
+          widget.onFocusChange?.call(_focused);
+        });
       });
     }
   }
 
-  void _handleShowHoverHighlight() {
-    widget.onShowHoverHighlight?.call(_hovering && widget.enabled && _canShowHighlight);
+  bool _shouldShowHoverHighlight({@required FocusableActionDetector target}) {
```

Comment:
Since these are both only called inside of `_mayTriggerCallback`, just move them to be local functions there (without their leading `_`).  They don't need to have a named argument: there's only one argument, and it's required.